libs/features/cdt/cdt.js

```diff
@@ -0,0 +1,123 @@
+    appendTimerBox(timerBlock, daysLeft, cdtDays);
+    appendSeparator(timerBlock);
+    appendTimerBox(timerBlock, hoursLeft, cdtHours);
+    appendSeparator(timerBlock);
+    appendTimerBox(timerBlock, minutesLeft, cdtMins);
+  }
+
+  function appendTimerLabel(parent, label) {
+    createTag('div', { class: 'timer-label' }, label, { parent });
+  }
+
+  function removeCountdown() {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function render(daysLeft, hoursLeft, minutesLeft) {
+    if (!isVisible) return;
+
+    removeCountdown();
+
+    appendTimerLabel(container, cdtLabel);
+    appendTimerBlock(container, daysLeft, hoursLeft, minutesLeft);
+  }
+
+  function updateCountdown() {
+    const instant = new URL(window.location.href)?.searchParams?.get('instant');
+    const currentTime = instant ? new Date(instant) : Date.now();
+
+    for (let i = 0; i < timeRangesEpoch.length; i += 2) {
+      const startTime = timeRangesEpoch[i];
+      const endTime = timeRangesEpoch[i + 1];
+
+      if (currentTime >= startTime && currentTime <= endTime) {
+        isVisible = true;
+        const diffTime = endTime - currentTime;
+        const daysLeft = Math.floor(diffTime / (1000 * 60 * 60 * 24));
+        const hoursLeft = Math.floor((diffTime % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+        const minutesLeft = Math.floor((diffTime % (1000 * 60 * 60)) / (1000 * 60));
+        render(daysLeft, hoursLeft, minutesLeft);
+        return;
+      }
+    }
+
+    isVisible = false;
+    clearInterval(interval);
+    removeCountdown();
+  }
+
+  function startCountdown() {
+    const oneMinuteinMs = 60000;
+    updateCountdown();
+    interval = setInterval(updateCountdown, oneMinuteinMs);
+  }
+
+  startCountdown();
+}
+
+function isMobile() {
+  return window.matchMedia('(max-width: 767px)').matches;
+}
+
+export default async function initCDT(el, classList) {
+  const placeholders = ['cdt-ends-in', 'cdt-days', 'cdt-hours', 'cdt-mins'];
+  const [cdtLabel, cdtDays, cdtHours, cdtMins] = await Promise.all(
+    placeholders.map((placeholder) => replaceKey(placeholder, getConfig())),
+  );
+
+  const cdtMetadata = getMetadata('countdown-timer');
+  if (cdtMetadata === null) {
+    throw new Error('Metadata for countdown-timer is not available');
+  }
+
+  const cdtRange = cdtMetadata.split(',');
+  if (cdtRange.length % 2 !== 0) {
+    throw new Error('Invalid countdown timer range');
+  }
+
+  const timeRangesEpoch = cdtRange.map((time) => {
+    const parsedTime = Date.parse(time?.trim());
+    return Number.isNaN(parsedTime) ? null : parsedTime;
+  });
+  if (timeRangesEpoch.includes(null)) {
+    throw new Error('Invalid format for countdown timer range');
+  }
+
+  const cdtDiv = createTag('div', { class: 'countdown-timer' }, null, { parent: el });
+  cdtDiv.classList.add(isMobile() ? 'vertical' : 'horizontal');
+  cdtDiv.classList.add(classList.contains('dark') ? 'dark' : 'light');
```

Comment:
shouldn't the `light` theme be default? We would have _default_ + _dark_ theme. If you don't specify a theme there's no default one and the counter will look unstyled.
<img width="352" alt="Screenshot 2024-09-23 at 15 19 24" src="https://github.com/user-attachments/assets/b5a0e607-4794-4984-ae6c-7b5837ed4366">